File: 15_Task(Sort)/DP_SortCollections/DP_SortCollections/Vector.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DP_SortCollections
{
    class Vector<T>: ISortable<T>
    {
        public T[] Array { get; set; }

        public SortStrategy<T> SortStrategy { get; set; }

        public Vector() { }

        public Vector(int size, SortStrategy<T> sortStrategy)
        {
            SortStrategy = sortStrategy;

            Array = new T[size];
        }

        public T this[int index]
        {
            get => Array[index]; 
            set => Array[index] = value; 
        }

        public int Length { get => Array.Length; }

        public void Sort()
        {
            SortStrategy?.Sort(this);
        }

        public void Print()
        {
            foreach (var item in Array)
            {
                Console.Write($"{item} ");
            }
            Console.WriteLine();
        }
    }
}

File: 15_Task(Sort)/HomeWork/src/Collection/Vector.java
package Collection;

import Sort.Sort;

public class Vector<T> extends Collection<T> {

  T[] array;

  public Vector(Sort<T> sortStrategy, T[] collection) {
    super(sortStrategy, collection);
    array = collection;
  }

  @Override
  public T get(int i) {
    return array[i];
  }

  @Override
  public void set(int i, T value) {
    array[i] = value;
  }

  @Override
  public int count() {
    return array.length;
  }

  @Override
  public void print() {
    for (T item : array) {
      System.out.print(item + "  ");
    }
    System.out.println("\n");
  }
  
}
